### ISSUE TITLE
Question: effective batch size for MLM?

### ISSUE THREAD

**Issue by PhilipMay** [UKPLab/sentence-transformers#1778]
@nreimers your `per_device_train_batch_size` is 64 here. But how many GPUs did you use?

Do you have any experience with useful batch sizes here?

The paper "Don’t Stop Pretraining: Adapt Language Models to Domains and Tasks" is talking about a

> effective batch size of 2048 

That would mean to use 8 GPUs and Gradient Accumulation of 4 steps?

https://github.com/UKPLab/sentence-transformers/blob/b86eec31cf0a102ad786ba1ff31bfeb4998d3ca5/examples/unsupervised_learning/MLM/train_mlm.py#L23

**Comment by nreimers** [comment]
Typically larger batch sizes and a larger LR allows quicker training. The paper BERT on an academic budget gives quite a good overview.

**Comment by PhilipMay** [comment]
Ok. Thanks!
